docs/how-to/push-update-on-mainnet.md

```diff
@@ -0,0 +1,187 @@
+Overview
+========
+
+Software update is a protocol mechanism that allows nodes to agree on software changes without altering protocol constants and update to this software version.
+1. Software update is proposed. Proposal is a datatype that gets into
+   the blockchain. It contains information about version changes and
+   hashes of update files.
+2. Update files are uploaded to the S3 bucket.
+3. Software update is confirmed by voting from majority of nodes.
+4. Nodes that see an update try to download and apply it
+   automatically.
+
+Now, again, step-by-step.
+
+Prerequisites
+=============
+
+0. For better understanding of update system, read cardanodocs:
+  * https://cardanodocs.com/cardano/update-mechanism/
+  * https://cardanodocs.com/technical/updater/
+1. Cluster nodes' keys. They are crucial to have since cluster nodes
+   are the only nodes that have stake in the system so they are the
+   only nodes that can propose an update and vote for it. It's enough
+   to have keys for majority of stake. In mainnet we have 7 core nodes
+   with equal stake, so 4 keys should be enough.
+2. Access to the S3 bucket
+3. Installers corresponding to software version we're pushing to cluster
+   - Those are two executables (exe for win, pkg for mac) which are to
+     be provided by QA (after QA procedures passed). They are
+     installers provided by our CI.
+   - Advice: after providing installers, ask QA to confirm hashes are
+     same, we don't want to ship users incorrect installers to users.
+4. Software version
+   - Single integer, must be greater by 1 than the last confirmed
+     version. Initially `csl-daedalus` application has version 0. The
+     first update should have version 1, the next one 2, etc. If you
+     try to propose an update with version 3 while the last confirmed
+     one is 1, your update proposal won't be accepted by the nodes.
+   - Node knows its software version from `configuration.yaml` file,
+     `applicationVersion` field. When you propose an update to version
+     X, make sure that all attached installers use configuration with
+     `applicationVersion` set to `X`.
```

Comment:
CSL-1758 describes a bad infinite update loop scenario if we get the versions wrong. If we conclude in that issue that no technical safeguards will be implemented, and that human verification alone is acceptable then I suggest the following:

1. Swap items 3 and 4
2. In what's currently number 3, add that getting this wrong can result in Daedalus users suffering denial of service due to repeating Daedalus updates.

Reply:
@jmitchell done. Though as fersel said in the YT issue, this scenario is mostly improbable.